tests/infra_test.go

```diff
@@ -104,6 +104,57 @@ var _ = Describe("Infrastructure", func() {
 				Expect(stdout).To(ContainSubstring("go_goroutines"))
 			}
 		})
+
+		It("should throttle the Prometheus metrics access", func() {
+			By("Creating the VirtualMachineInstance")
+			vmi := tests.NewRandomVMI()
+
+			By("Starting a new VirtualMachineInstance")
+			obj, err := virtClient.RestClient().Post().Resource("virtualmachineinstances").Namespace(tests.NamespaceTestDefault).Body(vmi).Do().Get()
+			Expect(err).ToNot(HaveOccurred(), "Should create VMI")
+
+			By("Waiting until the VM is ready")
+			nodeName := tests.WaitForSuccessfulVMIStart(obj)
+
+			By("Finding the prometheus endpoint")
+			pod, err := kubecli.NewVirtHandlerClient(virtClient).ForNode(nodeName).Pod()
+			Expect(err).ToNot(HaveOccurred(), "Should find the virt-handler pod")
+
+			By("Scraping the Prometheus endpoint")
+
+			// this looks quite high, but it is hard to stress the monitoring code in a reliable manner without injecting delays.
+			concurrency := 300
+			errors := make(chan error)
+			for ix := 0; ix < concurrency; ix++ {
+				go func() {
+					_, _, err := tests.ExecuteCommandOnPodV2(virtClient,
+						pod, "virt-handler",
+						[]string{
+							"curl",
+							"-m",
+							"0.1",
+							"-L",
+							"-k",
+							fmt.Sprintf("https://%s:%s/metrics", pod.Status.PodIP, "8443"),
+						})
+					errors <- err
+				}()
+			}
+
+			errorCount := 0
+			for ix := 0; ix < concurrency; ix++ {
+				err := <-errors
```

Comment:
TODO: add a comment to explain *which* error are we expecting. E.g. 50x HTTP code? timeout?

Reply:
What do we expect?

Reply:
We expect _any_ error to be occurred. Here I'm collecting the statuses from the concurrent clients - this loop is also meant to act as sync point (will document with a comment)